src/main/java/com/serenitydojo/Catalog.java

```diff
@@ -3,18 +3,97 @@
     }
 
     public static Catalog withItems(CatalogItem... catalogItems) {
-        throw new RuntimeException("TODO, create catalog and add items to the list of available fruits");
-        // Catalog catalog = new Catalog();
-        // for (CatalogItem catalogItem : catalogItems) {
-        //     catalog.availableFruits.add(catalogItem);
-        // }
-        // return catalog
+        //throw new RuntimeException("TODO, create catalog and add items to the list of available fruits"); - Done
+        Catalog catalog = new Catalog();
+        for (CatalogItem catalogItem : catalogItems) {
+            catalog.availableFruits.add(catalogItem);
+        }
+        return catalog;
+    }
+
+    public List<CatalogItem> getAvailableFruits(){
+        //Ash to reviewer: Which one's preferred? Mutate the availableFruits List as I did here OR sort on streams instead and produce a new sorted List
+        availableFruits.sort(Comparator.comparing(catalogItem -> catalogItem.getFruit().name()));
+        return availableFruits;
+
+        //        Alternate solution: sort on streams instead and produce a new sorted List
+        //        List<CatalogItem> availableFruitsSorted =
+//                availableFruits.stream()
+//                .sorted(Comparator.comparing(catalogItem -> catalogItem.getFruit()))
+//                .collect(Collectors.toList());
+//
+//       return availableFruitsSorted;
+//        return availableFruits;
+    }
+    public void setUnitPriceOf(Fruit fruit, double unitPrice) {
+        //Ash: Commenting the exception now that I've created the map
+        //throw new RuntimeException("TODO, create a map to keep track of fruits and their prices");
+
+        // Ash to reviewer: To get the following preexisting line of code to work, I'd have to change the data type of the Map's key from Fruit to String
+        // fruitToPrice.put(fruit.name(), unitPrice);
+
+        fruitToUnitPrice.put(fruit, unitPrice);
+    }
+
+    public Double getUnitPriceOf(Fruit fruit) throws FruitUnavailableException{
+        Boolean isFruitAvailable = false;
+
+        // Ash to reviewer: Not sure in response to what exactly we wanted the "FruitUnavailableException", so I included it here too
+        checkFruitAvailability(fruit);
+        return fruitToUnitPrice.get(fruit);
+    }
+
+    public Double calculateFruitPriceForQuantity(Fruit fruit, Integer quantityInKilos) throws FruitUnavailableException{
+        Double price = getUnitPriceOf(fruit);
+        return quantityInKilos * price;
+    }
+
+    public Boolean checkEligibilityForDiscount(Integer quantityInKilos){
+        if (quantityInKilos >= 5){
+            return true;
+        }
+        return false;
+    }
+
+    public void markFruitEligibleForDiscount(Fruit fruit){
+        fruitsEligibleForDiscount.add(fruit);
+    }
+
+    public Boolean checkFruitAvailability(Fruit fruit) throws FruitUnavailableException {
+       Boolean isFruitAvailable = false;
```

Comment:
@globalworming I read up about "why boolean over Boolean". Do you suggest the primitive to lower memory usage (less expensive) or another reason? Would love to know.

Reply:
great that you found that out. one other reason is: you can have `Boolean b = null` but `boolean b` will always be initialized as `false` https://www.baeldung.com/java-primitives-vs-objects#4-default-values

as your code doesn't need to support a `null` case, using `boolean` is less complicated. otherwise someone might see that and think "wait, what am I missing here, is `isFruitAvailable` set to null anywhere?"